JWBestPracticeApps/FeedTableViewController/FeedTableViewController/FeedTableViewController.swift

```diff
@@ -0,0 +1,64 @@
+//
+//  FeedTableViewController.swift
+//  FeedTableViewController
+//
+//  Created by Amitai Blickstein on 6/26/22.
+//
+
+import UIKit
+
+class FeedTableViewController: UITableViewController {
+    private var viewModel = FeedViewModel.shared
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        viewModel.addMoreItems()
+        
+        // Register the custom cell view
+        let feedNib = UINib(nibName: viewModel.cellNibName, bundle: .main)
+        tableView.register(feedNib, forCellReuseIdentifier: viewModel.cellReuseIdentifier)
+        tableView.isPagingEnabled = true
+        tableView.rowHeight = view.bounds.inset(by: view.safeAreaInsets).height
+    }
+
+    // MARK: UITableViewDelegate implementation
+    
+    override func tableView(_ tableView: UITableView, didEndDisplaying cell: UITableViewCell, forRowAt indexPath: IndexPath) {
+        guard let cell = cell as? PlayerItemCell
+        else { return }
+        
+        // Pause a cell as it goes offscreen
+        cell.playerView.player.pause()
+    }
+    
+    override func tableView(_ tableView: UITableView, willDisplay cell: UITableViewCell, forRowAt indexPath: IndexPath) {
+        guard let cell = cell as? PlayerItemCell
+        else { return }
+        
+        // Play a cell as it becomes visible
+        cell.playerView.player.play()
+
+        // Add more rows when we hit the end.
+        if indexPath.row == viewModel.count - 1 {
+            viewModel.addMoreItems()
+            tableView.reloadData()
+            // For more efficiency, use `reloadRows(at:with:)` for the (new && visible) rows.
```

Comment:
Why not make this more efficient, and have this line say why you are loading these? This BPA wants to show them how to make the best possible infinite feed table in the simplest way possible. And honestly, a little detail like this is exactly why some customers have complained to us about v3 feed tables because of a "blip" in the responsiveness of the players.

Reply:
I reasoned that deciding when to reload which cells was a table implementation detail, requiring no special knowledge of the SDK. Not that I disagree. I'll incorporate this.